### PR TITLE
Added simple transient ProtectedRegion functionality

### PR DESCRIPTION
**Pull request by WeaselSqueezer**
Added transience property to ProtectedRegion as a flag to only be stored in memory and not saved.

The transience check in isDirty() is needed so saveChanges() does not include transient regions.
This is tested and working as expected.

### CODE REVIEW

**Comment by WeaselSqueezer** [comment]
@sk89q are you able to take a look at this PR? it would be great if this can be in the next beta release, or at least in the full v6 release. Thanks! :smiley:

File: src/main/java/com/sk89q/worldguard/protection/regions/GlobalProtectedRegion.java [hunk 1]
```diff
@@ -38,11 +38,22 @@
 
     /**
      * Create a new instance.
+     * transience will be set to false, and this region can be saved.
```
Comment (kenzierocks) [inline src/main/java/com/sk89q/worldguard/protection/regions/GlobalProtectedRegion.java]: These docs don't look quite right. Can you reformat them?

File: src/main/java/com/sk89q/worldguard/protection/regions/ProtectedCuboidRegion.java [hunk 2]
```diff
@@ -39,13 +39,26 @@
 
     /**
      * Construct a new instance of this cuboid region.
+     * transience will be set to false, and this region can be saved.
```
Comment (kenzierocks) [inline src/main/java/com/sk89q/worldguard/protection/regions/ProtectedCuboidRegion.java]: Same here.

File: src/main/java/com/sk89q/worldguard/protection/regions/ProtectedRegion.java [hunk 3]
```diff
@@ -53,6 +53,7 @@
     protected BlockVector max;
 
     private final String id;
+    private final boolean transience;
```
Comment (kenzierocks) [inline src/main/java/com/sk89q/worldguard/protection/regions/ProtectedRegion.java]: Proposing name `memoryBased`.

**Comment by JonMcPherson** [comment]
Hey,
I just wanted to check up on this PR to see if it could be merged or fixed since it has been open for over a year.  This functionality would be a big help for many plugin developers as it would allow regions  created by other plugins to be managed and saved by those plugins rather than by WorldGuard.
Thanks for the consideration. I appreciate it.

**Comment by TomyLobo** [comment]
well, there's unresolved feedback

**Comment by WeaselSqueezer** [comment]
@TomyLobo ohhh oops. I made those javadoc changes on my local back in June but just forgot to push them to my remote.  This PR is now up to date with the commits for the unresolved javadoc feefback.
Let me know if there is anything else I need to change or fix for this PR.
Thanks!

**Comment by kenzierocks** [comment]
Could you do rebase this onto master? It seems you have some merge conflicts.

**Comment by WeaselSqueezer** [comment]
@kenzierocks Well I merged master and resolved the merge conflict using 'theirs', then applied the needed changes after the merge commit, so everything is in place. I am not sure why the Travis build failed though.

**Comment by WeaselSqueezer** [comment]
Ok I think now all feedback has been taken care of and this PR is ready to be merged.

Also note that for some reason, the GlobalProtectedRegion class had merge conflicts on commit 80b9601 for no reason I could think of considering the only change on master was the renaming (move) of the file.  So I resolved using the version from master, then made a new commit (bacb0ec) adding my changes back. But in doing that, for some reason git thinks the diffs are the entire file when in reality, the only changes are the new constructor and a few javadoc updates.
So that is why the diffs show the entire file and on parts that never changed.  This will have no effect on the merge though, and everything is in place for a successful merge.

Also, I think the Travis CI build failed due gradle issues, and it doesn't have anything to do with my branch which @kenzierocks confirmed in IRC

Let me know if anything else needs to be done before this can be merged. Thanks! :)

**Comment by WeaselSqueezer** [comment]
Can we take another look at this and consider merging it or at least implementing this functionality?  I want to release a plugin that requires this functionality, so it would be much appreciated if this could be considered.
thanks!

**Comment by wizjany** [comment]
can you squash/rebase?
i'd like to get this and #347 in once i'm sure 1.10 builds are stable for an experimental feature release.

**Comment by WeaselSqueezer** [comment]
@wizjany awesome.  ya I can try, but I made the mistake of merging my PR-335 branch into master and merging from canonical master a few times which would make a rebase difficult and messy as this PR is extremely stale.  Would it be better to just open a new clean PR?

**Pull request closed without merging**